mart/callbacks/adversary_connector.py

```diff
@@ -76,12 +112,18 @@ def model(input, target):
             batch = self.batch_c15n.revert(input, target)
 
             if hasattr(pl_module, "attack_step"):
+                # Users can implement LightningModule:attack_step() for generating adversarial examples.
                 outputs = pl_module.attack_step(batch, dataloader_idx)
             else:
-                # LightningModule must have "training_step".
-                # Disable logging if we have to reuse training_step() of the target model.
+                # If there is no attack_step(), we will borrow LightningModule:training_step() with some modifications.
+                #   1. Disable the training logging mechanism.
+                #   2. Switch to the training mode to get the loss value, except for children modules of BatchNorm and Dropout.
                 with MonkeyPatch(pl_module, "log", lambda *args, **kwargs: None):
-                    outputs = pl_module.training_step(batch, dataloader_idx)
+                    with training_mode(
```

Comment:
What is the use case here? Are you seeing train-specific code diverging from eval-specific code in some use case?

Reply:
Yes. Many model implementations return the prediction in the eval mode, and return the loss in the training mode.

In our use case, `anomalib test` runs the model in the eval mode, in which we won't get the loss.